modules/Bio/EnsEMBL/Variation/DBSQL/PublicationAdaptor.pm

```diff
@@ -226,7 +226,170 @@ sub fetch_all_by_Variation {
 
 }
 
+=head2 fetch_sources_by_pmid
 
+  Arg [1]    : String $pmid
+  Example    : $publication_adaptor->fetch_sources_by_pmid( 1234 );
+  Description: Retrieves a hashref of variants linked to publication sources via a pmid
+  Returntype : hashref
+  Exceptions : throw if pmid argument is not defined
+  Caller     : general
+  Status     : At Risk
+
+=cut
+
+sub fetch_sources_by_pmid {
+  my $self = shift;
+  my $pmid = shift;
+
+  throw("pmid must be defined") if(!defined($pmid));
+
+  my $pub = $self->fetch_by_pmid($pmid);
+  my $sources = $self->fetch_sources($pub->{dbID});
+
+  return $sources
+}
+
+=head2 fetch_sources_by_pmcid
+
+  Arg [1]    : String $pmcid
+  Example    : $publication_adaptor->fetch_sources_by_pmcid( 'PMC1234' );
+  Description: Retrieves a hashref of variants linked to publication sources via a pmcid
+  Returntype : hashref
```

Comment:
Add example for return type { variant_one =>'EPMC', variant_two => 'EPMC,dbSNP'}